### PR TITLE
Docker API Proxy Demo

### PR DESCRIPTION
**Pull request by devosray**
Initial implementation of the Docks API. Currently, it provides an HTTP endpoint that proxies all requests as well as all request body content to the Docker REST API.

### CODE REVIEW

**Comment by egeldenhuys** [comment]
Will test it now.

File: .idea/docks.iml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (egeldenhuys) [inline .idea/docks.iml]: Should files from the `.idea` directory be committed?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+
+## Development
+If you use the included Docker file for development, there is no need to install NodeJS on your local machine.
+1. `docker build -f Dockerfile-dev -t docks-development .`
+2. `docker run -it -v "$(pwd)"/:/app -p 8080:8080 -v /var/run/docker.sock:/var/run/docker.sock docks-development `
+
+When the development image runs (step 2), npm will check for any updates. The Express server will automatically
+reload as you edit local files.
+
+Not: currently, the development container will set the ownership of the `node_modules` folder and all installed
```
Comment (egeldenhuys) [inline README.md]: Confusing spelling error: Not -> Note

File: .idea/docks.iml (hunk 1 above)
Comment (devosray) [reply .idea/docks.iml]: I'm always including the bare minimum amount of IntelliJ project files to make it a valid project that can be easily imported but still have a small footprint. It should not interfere much.

I feel that IntelliJ is a common IDE among our team and making the project a real project would be a nice bonus. No personal data or static paths are committed. If you have strong feelings about it we can remove it.

File: README.md (hunk 2 above)
Comment (devosray) [reply README.md]: I'll correct it

File: .idea/docks.iml (hunk 1 above)
Comment (egeldenhuys) [reply .idea/docks.iml]: Then it is fine

**Pull request merged**